### PR TITLE
Update Tendermint version to v0.37.0

### PR DESCRIPTION
**Pull request by sergio-mena**
Part of #9091 

Updated Tendermint version on `v0.37.x`

---

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ var TMCoreSemVer = TMVersionDefault
 const (
 	// TMVersionDefault is the used as the fallback version of Tendermint Core
 	// when not using git describe. It is formatted with semantic versioning.
-	TMVersionDefault = "0.34.20"
```
Comment (sergio-mena) [inline version/version.go:8]: When backporting to v0.37.x, should this read `v0.37.0-alpha.1` ?

Comment (williambanfield) [reply version/version.go:8]: My intuition is to not have this this way in main. I _think_ we'd want the version in `main` to be something like `v0.38.0-dev` or similar to indicate that we are working on v0.38.

Comment (sergio-mena) [reply version/version.go:8]: Agree with `v0.38.0-dev`(I was thinking of doing it later on, by the time we create `v0.38.x`, but can be now as well).
Let me change the base of this PR...

Comment (sergio-mena) [reply version/version.go:10]: ... and I'll open another for `main`

File: version/version.go [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-)
+var TMCoreSemVer = TMVersionDefault
 
 const (
 	// TMVersionDefault is the used as the fallback version of Tendermint Core
 	// when not using git describe. It is formatted with semantic versioning.
-	TMVersionDefault = "0.34.20"
-	// ABCISemVer is the semantic version of the ABCI library
-	ABCISemVer = "0.18.0"
+	TMVersionDefault = "0.37.0"
```
Comment (thanethomson) [inline version/version.go:8]: Since the version we're going to cut will be `v0.37.0-alpha.1`:

```suggestion
	TMVersionDefault = "0.37.0-alpha.1"
```

**Review by thanethomson — COMMENTED** [review]
Thanks for this @sergio-mena!

Just one question regarding the GitHub Actions workflows updates (Dependabot, Mergify and the nightly E2E workflows): do they even do anything if committed to this branch? I was under the impression that those workflows on `main` are the only ones that GitHub pays attention to.

**Comment by sergio-mena** [comment]
> Just one question regarding the GitHub Actions workflows updates

@thanethomson that was me just screwing up a rebase... probably a git overdose over the summer :-D

Fixed now. BTW, I realise that this PR# is also covering the backport of #9341. I decided to leave it: we're changing contiguous lines, for we avoid useless conflicts

File: version/version.go (hunk 2 above)
Comment (sergio-mena) [reply version/version.go]: Absolutely

**Review by thanethomson — APPROVED** [review]

**Pull request merged**